hal/inc/socket_hal_posix.h

```diff
@@ -258,6 +260,11 @@ int sock_socket(int domain, int type, int protocol);
  */
 int sock_fcntl(int s, int cmd, ...);
 
+int sock_poll(struct pollfd* fds, nfds_t nfds, int timeout);
```

Comment:
Please also add doxygen documentation, since the rest of the functions in this header file are documented.

Reply:
Done